syntax = "proto3";
package object_detect;

service ObjectDetectService {
  rpc Detect (DetectRequest) returns (DetectResponse);
}

message DetectRequest {
  string image_base64 = 1;
  int32 user_id = 2;
  string target_item = 3;
}

message DetectResponse {
  string status = 1;  // JSON: {"type": "object_detection", "result": "PASS", ...}
  int32 user_id = 2;
}